staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -2114,6 +2116,12 @@ func (d *awsDisk) waitForAttachmentStatus(status string) (*ec2.VolumeAttachment,
 	// So we tolerate a limited number of failures.
 	// But once we see more than 10 errors in a row, we return the error
 	describeErrorCount := 0
+
+	// Attach/detach usually takes time. It does not make sense to start
+	// polling DescribeVolumes before some initial delay to let AWS
+	// process the request.
+	time.Sleep(getInitialAttachDetachDelay(status))
+
 	var attachment *ec2.VolumeAttachment
 
 	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
```

Comment:
any ideas for metrics that could help us tune this more in the future, if necessary? How did you figure out that 2s is average?

Reply:
I set really high frequency of DescribeVolumes polling and ran bunch of tests.

We still have metric `storage_operation_duration_seconds_sum{volume_plugin=" kubernetes.io/aws-ebs", operation_name="volume_attach"}`, but it includes `DescribeInstance` + `AttachVolume` calls + `DescribeVolumes` polling.